faf-commons-api/src/main/java/com/faforever/commons/api/dto/MapParams.java

```diff
@@ -0,0 +1,18 @@
+package com.faforever.commons.api.dto;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeInfo.Id;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@JsonTypeInfo(use = Id.NAME, property = "type")
+@JsonSubTypes(
+    @JsonSubTypes.Type(value = NeroxisGeneratorParams.class, name = "neroxis")
+)
+@Getter
+@Setter
+@NoArgsConstructor
+public abstract class MapParams {
```

Comment:
Use an interface instead of an abstract class and remove the ununused Lombok annotations